File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__lucky_number.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;
using static System.Math;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        int o = 1000000;
        int[] d = { 0, 1, -1 };
        int[] x = new int[n];

        foreach( int i in Range( 0, n ) ) {
            x[i] = int.Parse( ReadLine() );
        }
        foreach( int i in Range( 0, n ) ) {
            for( int j = i + 1; j < n; j++ ) {
                foreach( int k in Range( 0, 5 ) ) {
                    foreach( int l in Range( 0, 5 ) ) {
                        int n1 = 0;
                        int n2 = 0;

                        switch( k ) {
                        case 3:
                            n1 = int.Parse( $"1{x[i]}" );
                            break;
                        case 4:
                            n1 = int.Parse( $"{x[i]}1" );
                            break;
                        default:
                            n1 = x[i] + d[k];
                            break;
                        }
                        switch( l ) {
                        case 3:
                            n2 = int.Parse( $"1{x[j]}" );
                            break;
                        case 4:
                            n2 = int.Parse( $"{x[j]}1" );
                            break;
                        default:
                            n2 = x[j] + d[l];
                            break;
                        }
                        o = Min( o, Abs( n1 - n2 ) );
                    }
                }
            }
        }
        WriteLine( o );
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__lucky_number.py
# 実行時間 20ms
n = int(input())
o = 1000000
d = [0, 1, -1, "1"]
x = []

for _ in range(n):
    x.append(int(input()))
for i in range(n):
    for j in range(i + 1, n):
        for k in range(5):
            for l in range(5):
                n1 = 0
                n2 = 0

                if k == 3:
                    n1 = int(f"{d[3]}{x[i]}")
                elif k == 4:
                    n1 = int(f"{x[i]}{d[3]}")
                else:
                    n1 = x[i] + d[k]
                if l == 3:
                    n2 = int(f"{d[3]}{x[j]}")
                elif l == 4:
                    n2 = int(f"{x[j]}{d[3]}")
                else:
                    n2 = x[j] + d[l]
                o = min(o, abs(n1 - n2))
print(o)
